src/kaleidoscope/driver/leddriver/Base.h

```diff
@@ -0,0 +1,57 @@
+/* -*- mode: c++ -*-
+ * BaseLEDDriver -- LED hardware driver base class for Kaleidoscope
+ * Copyright (C) 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#ifndef CRGB
+#error cRGB and CRGB *must* be defined before including this header!
+#endif
+
+namespace kaleidoscope {
+namespace driver {
+namespace leddriver {
+
+struct BaseProps {
+  typedef int8_t LEDCountType;
```

Comment:
Is this whole abstraction so that we can have some keyboards with a led count type of uint8_t? If so, why not just make all of them uint8_t?  Or does the raise have more than 256 LEDs?

Reply:
We can't just use `uint8_t`, because we have code that does essentially `for (int8_t i = 0; i < led_count; i++)`, and with an `uint8_t`, that condition would never be false. We could have a separate check that looks at led count, but that's more PROGMEM the compiler can't optimize out. This is something we did to fix #385, in #432.

So we either figure something out to address those in a different way, or we need to stick with signed ints. In which case, we need `int16_t` for the Raise.

In any case, it's not hard to imagine a board with more than 256 LEDs, either: a full-size with both side- and underglow, and you're easily looking at more than that.